### PR TITLE
Telescope model - Issue with missing lon/lat

### PR DESCRIPTION
**Pull request by Theodlz**
This PR addresses https://github.com/skyportal/skyportal/issues/2970. I made some changes to the model and added comments to justify them. I'm tagging @stefanv (that opened the issue) as well as @mcoughlin that assigned me. 
The comments I left in the model have to be removed of course, they are here only for the reviewers to understand the changes I made. But maybe some of them could stay? (for example, I added a comment not to justify any of my changes but to clarify some bits of the code in the `def observer()`)

### CODE REVIEW

**Review by mcoughlin — APPROVED** [review]
@Theodlz Nice. Hopefully this finishes it off!

**Comment by Theodlz** [comment]
I noticed in the flaky tests that there is also a galaxy tests in the gcn evet flaky tests... Let me fix it before merging !

**Comment by Theodlz** [comment]
Before this goes in, maybe #3101 could go in first ? Its been ready for a some days now.

File: skyportal/models/telescope.py
Comment (guynir42) [inline skyportal/models/telescope.py:75]: Maybe we should not return anything, but instead set `self._observer = None` and then only at the very end return `self._observer`. That way this logic is only called once, and then the result is cached.

File: skyportal/models/telescope.py
Comment (guynir42) [inline skyportal/models/telescope.py:100]: If we cache the results of `self._observer` this message will only printout once. I think that's better than having it pop up all the time.

File: skyportal/models/telescope.py
Comment (guynir42) [inline skyportal/models/telescope.py:108]: Just so you know, you can also add comments on github (like reviewer comments but on your own code). 
That may be easier that adding code comments and then cleaning them up, but it is totally up to you either way.

**Review by guynir42 — APPROVED** [review]
Looks good. I would cache the result of `self.observer` in `self._observer` but otherwise the code looks fine. I am really surprised we are still dealing with this - I hope this takes care of it finally.

File: skyportal/models/telescope.py
Comment (Theodlz) [reply skyportal/models/telescope.py:104]: Ok well I did just that, i'll let you have a look @guynir42

File: skyportal/models/telescope.py
Comment (Theodlz) [reply skyportal/models/telescope.py]: I think all those if statement can be combined into one, where if triggered, we set the `self._observer` to None and return it, so that return will be triggered just once. The reason for keeping the return is that when running for the first time (when there are no observer yet) it will keep it from running the next try expect.

**Comment by Theodlz** [comment]
I hope this will help. As mentionned before, I never managed to reproduced locally the issues encountered in production, so it's hard to try it and give any guarantee that this will fix it. But it should make it more robust for sure.

**Review by mcoughlin — APPROVED** [review]
Thanks for all your effort here @Theodlz.

**Review by guynir42 — APPROVED** [review]

**Pull request merged**
